pkg/diagnostics/http_tracing.go

```diff
@@ -182,104 +184,20 @@ func tracestateToHeader(sc trace.SpanContext, setHeader func(string, string)) {
 	}
 }
 
-func getAPIComponent(apiPath string) (string, string) {
-	// Dapr API reference : https://docs.dapr.io/reference/api/
-	// example : apiPath /v1.0/state/statestore
-	if apiPath == "" {
-		return "", ""
-	}
-
-	// Split up to 4 delimiters in '/v1.0/state/statestore/key' to get component api type and value
-	tokens := strings.SplitN(apiPath, "/", 4)
-	if len(tokens) < 3 {
-		return "", ""
-	}
-
-	// return 'state', 'statestore' from the parsed tokens in apiComponent type
-	return tokens[1], tokens[2]
-}
-
 func spanAttributesMapFromHTTPContext(rw responsewriter.ResponseWriter, r *http.Request) map[string]string {
-	// Span Attribute reference https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions
-	path := r.URL.Path
-	method := r.Method
-	statusCode := rw.Status()
-
-	m := map[string]string{}
-	_, componentType := getAPIComponent(path)
-
-	var dbType string
-	switch componentType {
-	case "state":
-		dbType = stateBuildingBlockType
-		m[dbNameSpanAttributeKey] = rw.UserValueString("storeName")
-
-	case "secrets":
-		dbType = secretBuildingBlockType
-		m[dbNameSpanAttributeKey] = rw.UserValueString("secretStoreName")
-
-	case "bindings":
-		dbType = bindingBuildingBlockType
-		m[dbNameSpanAttributeKey] = rw.UserValueString("name")
-
-	case "invoke":
-		m[gRPCServiceSpanAttributeKey] = daprGRPCServiceInvocationService
-		targetID := rw.UserValueString("id")
-		m[netPeerNameSpanAttributeKey] = targetID
-		m[daprAPISpanNameInternal] = "CallLocal/" + targetID + "/" + rw.UserValueString("method")
-
-	case "publish":
-		m[messagingSystemSpanAttributeKey] = pubsubBuildingBlockType
-		m[messagingDestinationSpanAttributeKey] = rw.UserValueString("topic")
-		m[messagingDestinationKindSpanAttributeKey] = messagingDestinationTopicKind
+	// Init with a worst-case initial capacity
+	m := make(map[string]string, 7)
```

Comment:
I don't think it needs a variable because it's something used by the compiler only, but it does need a clearer comment.

Essentially, by doing `make(map[string]string, 7)` we are telling the compiler to allocate the map with an initial capacity for (up to) 7 items. If we don't pass it explicitly, the compiler will allocate an initial capacity as it sees fit. Because in the code below we do not store more than 7 unique keys in the map, this makes it possible to avoid re-allocations (which happen when the map was allocated with a smaller capacity than needed). Re-allocations take time and also add more work on the garbage collector.

The number 7 is just counting the number of unique keys used. Nothing really "magic" about it, and setting it to 6 or 8 would not cause anything to break (although could cause inefficiencies).

I'll add a comment in a bit to clarify.

Reply:
Okay, adding a comment is sufficient

Reply:
Done